lib/pluggy_elixir/config.ex

```diff
@@ -0,0 +1,83 @@
+defmodule PluggyElixir.Config do
+  @moduledoc """
+  PluggyElixir client configurations
+
+  To use `PluggyElixir` you must add into your config file `config.exs` the
+  required config:
+
+      config :pluggy_elixir,
+        client_id: "your-app-client-id",
+        client_secret: "your-app-client-secret",
+
+
+  Allowed configuration:
+
+  - client_id(required): A client id provide by Pluggy
+  - client_secret(required): A client secret provide by Pluggy
+  - non_expiring_api_key: A boolean value to enable create a non expiring API
+  KEY. (default is `false`)
+  - sandbox: A boolean value to enable using sandbox `Connector`. (default is `false`)
+  - host: A string containing the API host. (default is `"api.pluggy.ai"`)
+
+  """
+
+  @default_host "api.pluggy.ai"
+
+  @doc false
+  def get_client_id, do: get_pluggy_elixir_config(:client_id)
+
+  @doc false
+  def get_client_secret, do: get_pluggy_elixir_config(:client_secret)
+
+  @doc false
+  def get_host_uri do
+    ~r"(?<scheme>http[s]?)?(:[\/]{2})?(?<host>[\w\.]+)(:?(?<port>\d+))?(\/(?<path>.*))?"
+    |> Regex.named_captures(get_pluggy_elixir_config(:host, @default_host))
+    |> to_uri()
+  end
```

Comment:
Why not use something like `URI.parse`?
```elixir
iex>URI.parse("https://brainn.co") 
%URI{
  authority: "brainn.co",
  fragment: nil,
  host: "brainn.co", 
  path: nil,
  port: 443,
  query: nil,
  scheme: "https",
  userinfo: nil
}
```

Reply:
Because `URI.parse/1` can parse only well formated urls. A `"localhos"` url is parsed as:
```elixir
iex(1)> URI.parse "localhost"
%URI{
  authority: nil,
  fragment: nil,
  host: nil,
  path: "localhost",
  port: nil,
  query: nil,
  scheme: nil,
  userinfo: nil
}

Reply:
it is parsed correctly when there is the scheme

```elixir
URI.parse("http://localhost")
%URI{
  authority: "localhost",
  fragment: nil,
  host: "localhost",
  path: nil,
  port: 80,
  query: nil,
  scheme: "http",
  userinfo: nil
}
```

Reply:
yes, but the idea is the configuration accepts both, just domain and full url.